pkg/apis/monitoring/v1/types.go

```diff
@@ -813,11 +813,15 @@ type PrometheusRuleSpec struct {
 // upstream Prometheus struct definitions don't have json struct tags.
 
 // RuleGroup is a list of sequentially evaluated recording and alerting rules.
+// Note: PartialResponseStrategy is only used by ThanosRuler and will
+// be ignored by Prometheus instances.  Valid values for this field are 'warn'
```

Comment:
I don't see us ignoring the field. Are we sure Prometheus wouldn't refuse to load the rules?

Reply:
Indeed Prometheus will reject rules with the `partial_response_strategy` field. But AFAICT the field is emptied when generating rules for Prometheus: https://github.com/coreos/prometheus-operator/blob/ca852a3d20f8051c62383d3858e9dca622ddf7ac/pkg/prometheus/rules.go#L213

Reply:
ack

Reply:
this seems somewhat risky, I would want us to make sure that we create a copy of the rule before we set something like this, as we are modifying the item in the cache with this line

Reply:
cc @pgier ^ we should do this in a follow up